static_frame/core/index_level.py

```diff
@@ -372,6 +372,37 @@ def get_labels(index: Index,
                 next_depth = depth + 1
                 levels.extend((lvl, next_depth) for lvl in level.targets)
 
+    def relabel_at_specific_depth(self,
+            mapper: RelabelInput, depth_level: int
+            ) -> 'IndexLevel':
+        '''
+        Returns a new IndexLevel instance with relabeled indices at the specified `depth_level` according to `mapper`.
+
+        It will recursive through the internal index levels to apply a `relabel` to all the appropriate indices.
+        '''
+        if depth_level == 0:
+            return self.__class__(
+                index=self.index.relabel(mapper),
+                targets=self.targets,
+                offset=self.offset,
+                own_index=True,
+                depth_reference=self._depth
+            )
+
+        if self.targets is None:
+            raise ValueError(f'cannot relabel at depth {depth_level}')
+
+        new_targets = np.empty(len(self.targets), dtype=object)
```

Comment:
We have a constant for this:
```suggestion
        new_targets = np.empty(len(self.targets), dtype=DTYPE_OBJECT)
```